### PR TITLE
Dockerfile updates

### PR DESCRIPTION
**Pull request by lazzarello**
* Add solc binary to Dockerfile
* Run container as root
* Document interactive usage of container

---
This change is [Reviewable](https://reviewable.io/reviews/trailofbits/manticore/882)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -119,6 +119,33 @@ def hook(state):
...
+`docker build -t manticore .`
+
+From there it can be run interactively with examples mounted with the following
+
+``docker run -it -v `pwd`/examples:/root/examples manticore``
+
+This drops you into a shell in the container. The following is a run of the Getting Started example.
+
+```
+2blockchains:manticore leeazzarello$ docker run -it -v `pwd`/examples:/root/examples manticore
```
[the remark this answers is not in the document]
Comment (lazzarello) [reply README.md]: sure...copy pasta...

[the remark this answers is not in the document]
Comment (lazzarello) [reply README.md]: pushed

**Comment by mossberg** [comment]
thanks for the PR, looking at it now (sorry for taking so long!). thanks also for the docker instructions; i don't have a ton of docker experience so they're super helpful :)

File: Dockerfile [hunk 2]
```diff
@@ -7,9 +7,12 @@ RUN apt-get -y update && \
...
-RUN useradd -m manticore
-USER manticore
-WORKDIR /home/manticore
-ENV HOME /home/manticore
+RUN apt-get install -y build-essential software-properties-common && \
+    add-apt-repository -y ppa:ethereum/ethereum && \
+    apt-get update && \
+    apt-get install -y solc ethereum
+
+WORKDIR /root
```
Comment (mossberg) [inline Dockerfile]: any particular reason to change from the manticore user to the root user?

**Comment by mossberg** [comment]
I just added a commit with some superficial, stylistic README changes. Good to merge once we figure out the root/manticore user thing :)

File: Dockerfile [hunk 3]
```diff
@@ -7,9 +7,12 @@ RUN apt-get -y update && \
...
 	pip install .
 
-RUN useradd -m manticore
-USER manticore
-WORKDIR /home/manticore
-ENV HOME /home/manticore
+RUN apt-get install -y build-essential software-properties-common && \
+    add-apt-repository -y ppa:ethereum/ethereum && \
+    apt-get update && \
+    apt-get install -y solc ethereum
```
Comment (mossberg) [inline Dockerfile]: what does installing the `ethereum` package get us?

Comment (mossberg) [reply Dockerfile]: i noticed that it does add some time to the docker build stage, but then i realized docker hub will generally be doing that, not our users directly, so i don't think it's a big deal, nvm.

**Comment by mossberg** [comment]
I ended up reverting the changes related to root user, and reverted back to the manticore user, but we can have future discussion for a later PR if need be :)

**Pull request merged**